File: codigos_exercicios/python/exercicio23.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Exercicio 23
"""

import numpy as np
import scipy.optimize as opt

# definindo o sistema
myfun = lambda x : [2*x[0] - x[1] - np.exp(-x[1]), -x[0] + 2*x[1] - np.exp(-x[1])]

# resolvendo
x = opt.fsolve(myfun,[-5,5])

File: codigos_exercicios/exercicio23.m
% definindo o sistema
myfun = @(x) [2*x(1) - x(2) - exp(-x(1));
    -x(1) + 2*x(2) - exp(-x(2))];

% estimativa inicial
x0 = [-5; -5];

% opcao para mostrar o output
options = optimoptions('fsolve','Display','iter');

% resolvendo
[x, valor_funcao] = fsolve(myfun,x0,options)